layouts/community/ergodox/bepo/keymap.c

```diff
@@ -0,0 +1,455 @@
+ *                                      |      |      |      |      |      |
+ *                               ,------|------|------|      |------+------+------.
+ *                               |      |      |      |      |      |      |      |
+ *                               |      |      |------|      |------|      |      |
+ *                               |      |      |      |      |      |      |      |
+ *                               `--------------------'      `--------------------'
+ */
+[QWER] = LAYOUT_ergodox_pretty(
+_______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,
+_______, BP_Q,    BP_W,    BP_E,    BP_R,    BP_T,    _______,          _______, BP_Y,    BP_U,    BP_I,    BP_O,    BP_P,    _______,
+_______, BP_A,    BP_S,    BP_D,    BP_F,    BP_G,                               BP_H,    BP_J,    BP_K,    BP_L,    _______, _______,
+_______, BP_Z,    BP_X,    BP_C,    BP_V,    BP_B,    _______,          _______, BP_N,    BP_M,    _______, _______, _______, _______,
+_______, _______, _______, _______, _______,                                              _______, _______, _______, _______, _______,
+                                             _______, _______,          _______, _______,
+                                                      _______,          _______,
+                                    _______, _______, _______,          _______, _______, _______
+),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        // keypad "double 0"
+        case KP_00:
+            if (record->event.pressed) {
+                SEND_STRING(SS_TAP(X_KP_0) SS_DOWN(X_KP_0));
+            } else {
+                SEND_STRING(SS_UP(X_KP_0));
+            }
+            return false;
+        // add a momentary layer switch to the Mod-Tap ESC/LCTL key so that bepo users can use qwerty-like ctrl key bindings
+        case LCTL_T(KC_ESC):
+            if (record->event.pressed) {
+                layer_on(QWER);
+            } else {
+                layer_off(QWER);
+            }
+            return true;
+    }
+    return true;
+};
+
+void matrix_init_user (void) {
+    ergodox_board_led_off();
+    ergodox_right_led_1_off();
+    ergodox_right_led_2_off();
+    ergodox_right_led_3_off();
+}
+
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_NUM_LOCK)) {
+        ergodox_right_led_1_set(8);
+        ergodox_right_led_1_on();
+    } else {
+        ergodox_right_led_1_off();
+    }
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        ergodox_right_led_2_set(2);
+        ergodox_right_led_2_on();
+    } else {
+        ergodox_right_led_2_off();
+    }
+    if (usb_led & (1<<USB_LED_SCROLL_LOCK)) {
+        ergodox_right_led_3_set(6);
+        ergodox_right_led_3_on();
+    } else {
+        ergodox_right_led_3_off();
+    }
+}
```

Comment:
```suggestion
bool led_update_user(led_t led_state) {
    ergodox_right_led_1_off();
    ergodox_right_led_2_off();
    ergodox_right_led_3_off();

    if (led_state.num_lock) {
        ergodox_right_led_1_set(8);
        ergodox_right_led_1_on();
    }
    if (led_state.caps_lock) {
        ergodox_right_led_2_set(2);
        ergodox_right_led_2_on();
    }
    if (led_state.scroll_lock) {
        ergodox_right_led_3_set(6);
        ergodox_right_led_3_on();
    }

    return false;
}
```

Reply:
Committed suggested changes. What's the rationale behind calling `*_off()` unconditionally instead of inside `else` statements?

Reply:
It's a bit less complex, and easier to read.  And it doesn't create any visual artifacts.